src/main/java/com/google/cloud/genomics/dataflow/pipelines/CountReads.java

```diff
@@ -0,0 +1,182 @@
+ * APIs OR BAM files and invoking GATK tools.
+ */
+public class CountReads {
+  private static final Logger LOG = Logger.getLogger(CountReads.class.getName());
+  private static CountReadsOptions options;
+  private static Pipeline p;
+  private static GenomicsFactory.OfflineAuth auth;
+  public static interface CountReadsOptions extends GenomicsDatasetOptions, GCSOptions {
+    @Description("The ID of the Google Genomics ReadGroupSet this pipeline is working with. "
+        + "Default (empty) indicates all ReadGroupSets.")
+    @Default.String("")
+    String getReadGroupSetId();
+
+    void setReadGroupSetId(String readGroupSetId);
+
+    @Description("The path to the BAM file to get reads data from.")
+    @Default.String("")
+    String getBAMFilePath();
+
+    void setBAMFilePath(String filePath);
+    
+    @Description("Whether to shard BAM reading")
+    @Default.Boolean(true)
+    boolean getShardBAMReading();
+
+    void setShardBAMReading(boolean newValue);
+    
+    class ContigsFactory implements DefaultValueFactory<Iterable<Contig>> {
+      @Override
+      public Iterable<Contig> create(PipelineOptions options) {
+        return Iterables.transform(Splitter.on(",").split(options.as(CountReadsOptions.class).getReferences()),
+            new Function<String, Contig>() {
+              @Override
+              public Contig apply(String contigString) {
+                ArrayList<String> contigInfo = newArrayList(Splitter.on(":").split(contigString));
+                return new Contig(contigInfo.get(0), 
+                    contigInfo.size() > 1 ? 
+                        Long.valueOf(contigInfo.get(1)) : 0, 
+                    contigInfo.size() > 2 ?
+                        Long.valueOf(contigInfo.get(2)) : -1);
+              }
+            });
+      }
+    }
+    
+    @Default.InstanceFactory(ContigsFactory.class)
+    @JsonIgnore
+    Iterable<Contig> getContigs();
+    
+    void setContigs(Iterable<Contig> contigs);
+  }
+  
+  public static void main(String[] args) throws GeneralSecurityException, IOException {
+    options = PipelineOptionsFactory.fromArgs(args).withValidation().as(CountReadsOptions.class);
+    GenomicsOptions.Methods.validateOptions(options);
+    auth = GCSOptions.Methods.createGCSAuth(options);
+    p = Pipeline.create(options);
+    DataflowWorkarounds.registerGenomicsCoders(p);
+
+    PCollection<Read> reads = getReads();
+    PCollection<Long> readCount = reads.apply(Count.<Read>globally());
+    PCollection<String> readCountText = readCount.apply(ParDo.of(new DoFn<Long, String>() {
+      @Override
+      public void processElement(DoFn<Long, String>.ProcessContext c) throws Exception {
+        c.output(String.valueOf(c.element()));
+      }
+    }).named("toString"));
+    readCountText.apply(TextIO.Write.to(options.getOutput()).named("WriteOutput"));
+    p.run();
+  }
+
+  private static PCollection<Read> getReads() throws IOException {
```

Comment:
It would be a little clearer to structure this so the set of contigs we're reading is passed in to getReads, IMHO, but that may require some refactoring of the dataflow-java utils.